src/data/list/cycle.lean

```diff
@@ -436,7 +436,22 @@ instance : has_coe (list α) (cycle α) := ⟨quot.mk _⟩
 @[simp] lemma mk'_eq_coe (l : list α) :
   quotient.mk' l = (l : cycle α) := rfl
 
-instance : inhabited (cycle α) := ⟨(([] : list α) : cycle α)⟩
+/-- The unique empty cycle. -/
+def nil : cycle α := (([] : list α) : cycle α)
+
+instance : has_emptyc (cycle α) := ⟨nil⟩
+
+instance : inhabited (cycle α) := ⟨nil⟩
+
+theorem cases_on (s : cycle α) : s = nil ∨ ∃ a (l : list α), s = a :: l :=
```

Comment:
This would be better defined as a recursor, so that the type resembles list.cases_on

Reply:
What's the difference? I'm not really sure how recursors work.

Reply:
Any better?